Gems/Atom/Feature/Common/Code/Source/PostProcessing/TaaPass.cpp

```diff
@@ -129,21 +130,27 @@ namespace AZ::Render
         m_accumulationAttachments[0] = FindAttachment(Name("Accumulation1"));
         m_accumulationAttachments[1] = FindAttachment(Name("Accumulation2"));
 
-        bool hasAttachments = m_accumulationAttachments[0] || m_accumulationAttachments[1];
-        AZ_Error("TaaPass", hasAttachments, "TaaPass must have Accumulation1 and Accumulation2 ImageAttachments defined.");
-
-        if (hasAttachments)
+        bool attachmentsValid = true;
+        // Make sure the attachments have images when the pass first loads.
+        for (auto i : { 0, 1 })
         {
-            // Make sure the attachments have images when the pass first loads.
-            for (auto i : { 0, 1 })
+            if (m_accumulationAttachments[0] && !m_accumulationAttachments[i]->m_importedResource)
+            {
+                attachmentsValid &= UpdateAttachmentImage(i);
```

Comment:
Bitwise operation is not recommended for bool. It may cause different behavior from logical operation in some case like 1&2=0. And it doesn't have the ability to do short circuit, in case when `attachmentsValid` is false, but you can break in the else branch.

Reply:
removed the bitwise operations, and break early if one attachment is invalid